eclair-core/src/main/scala/fr/acinq/eclair/db/Databases.scala

```diff
@@ -241,19 +244,19 @@ object Databases extends Logging {
 
       databases
     }
+  }
 
-    private def checkIfDatabaseUrlIsUnchanged(url: String, urlFile: File): Unit = {
-      def readString(path: Path): String = Files.readAllLines(path).get(0)
+  private def checkIfDatabaseUrlIsUnchanged(url: String, urlFile: File): Unit = {
+    def readString(path: Path): String = Files.readAllLines(path).get(0)
 
-      def writeString(path: Path, string: String): Unit = Files.write(path, java.util.Arrays.asList(string))
+    def writeString(path: Path, string: String): Unit = Files.write(path, java.util.Arrays.asList(string))
 
-      if (urlFile.exists()) {
-        val oldUrl = readString(urlFile.toPath)
-        if (oldUrl != url)
-          throw JdbcUrlChanged(oldUrl, url)
-      } else {
-        writeString(urlFile.toPath, url)
-      }
+    if (urlFile.exists()) {
+      val oldUrl = readString(urlFile.toPath)
+      if (oldUrl != url)
+        throw JdbcUrlChanged(oldUrl, url)
```

Comment:
Do we want to slightly update this to make the transition from `sqlite` to `postgres` more user friendly?
Right now the error they would get would be "The database URL has changed since the last start. It was `sqlite`, now it's `<some_postgres_url>`".

Maybe we could add a condition:

```
if (oldUrl != url) {
    if (oldUrl == "sqlite" || url == "sqlite") throw DbTypeChanged(<something>)
    else throw JdbcUrlChanged(oldUrl, url)
}
```

Where `DbTypeChanged` would have a message saying that switching from one type of DB to another is dangerous, make sure you have migrated the data manually first, etc?

Reply:
I went with something in-between in 9cb7edbab. I kept the existing exception, otherwise it forced us to list all possible cases, which was tedious, error prone, and not really useful: even when moving from postgres to postgres, the migration warning applies.